### DISCUSSION TITLE
I am a Newbie in open sources

### DISCUSSION THREAD

**Discussion by hJava72** [community/community#53772] [Programming Help]
Hello everybody!...I am a newbie in open source area.
In the past I have created desktop-based applications using the c, clipper, visual basic programming language. And now I am learning python, so please guide me so that I can contribute to the world of open sources.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @hJava72 ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion. We have a thread already for [Python resources](https://github.com/orgs/community/discussions/51390) you may find helpful. 

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by RamtinKosari** [comment]
Hi dear @hJava72 
That's great to hear that you're interested in contributing to the world of open source! Python is a versatile and widely used programming language, and there are numerous open source projects where you can contribute your skills. Here are some steps to get started:

Explore open source projects: Start by exploring popular open source projects on platforms like GitHub, GitLab, or Bitbucket. Look for projects that align with your interests or areas of expertise. You can search for projects by topic or language (in this case, Python).

Understand the project: Once you find a project that interests you, take some time to understand its purpose, goals, and the technologies it uses. Read the project's documentation, review its issue tracker, and look at the existing codebase to get a sense of the project's structure and coding style.

Start small: As a beginner, it's a good idea to start with small and manageable tasks. Look for issues labeled as "beginner-friendly," "good first issue," or similar tags. These issues are usually well-suited for newcomers and provide a way to get familiar with the project's codebase and development workflow.

Set up the development environment: Follow the project's documentation to set up the development environment. This typically involves installing dependencies, setting up a version control system, and configuring any necessary tools or frameworks.

Contribute and communicate: Once you've selected an issue to work on, start coding! Follow the project's contribution guidelines and coding conventions. Make sure to communicate with the project maintainers or the community to ask questions, seek clarifications, or get feedback on your work. This could be through project-specific communication channels such as mailing lists, forums, or chat platforms.

Submit your contribution: Once you're confident in your changes, submit a pull request or patch to the project. Provide a clear description of your changes and explain how they address the issue. Be prepared to receive feedback from the project maintainers and iterate on your code if necessary.

Learn from feedback: Open source is a collaborative process, and receiving feedback on your contributions is valuable. Take feedback positively and use it as an opportunity to learn and improve your skills. Engage in discussions, ask questions, and be open to suggestions from more experienced contributors.

Remember, contributing to open source is not just about writing code but also about being part of a community. Engage with the project's community, attend virtual meetups or conferences, and contribute in non-code ways such as documentation, testing, or helping with issue triage.

Lastly, be patient and persistent. Open source contributions take time and effort, but they can be rewarding and contribute to your personal and professional growth. Good luck on your open source journey!
